### PR TITLE
RATIS-1636. Support re-config ratis properties

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/projects/RATIS/issues/RATIS-1636

### CODE REVIEW

File: ratis-common/src/main/java/org/apache/ratis/conf/ReconfigurationBase.java [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+  public Collection<PropertyChange> getChangedProperties(
+      RaftProperties newConf, RaftProperties oldConf) {
+    return reconfigurationUtil.parseChangedProperties(newConf, oldConf);
+  }
+
+  /**
+   * A background thread to apply property changes.
+   */
+  private static class ReconfigurationThread extends Thread {
```
Comment (maobaolong) [inline ratis-common/src/main/java/org/apache/ratis/conf/ReconfigurationBase.java:105]: You can use call event trigger the property change event and handle all property change in one method.

**Comment by szetszwo** [comment]
@codings-dan , thanks for working on this.  I am reviewing the change.

**Review by szetszwo — COMMENTED** [review]
@codings-dan , thanks a lot for working on this!  I suggest to do some refactoring; see https://issues.apache.org/jira/secure/attachment/13054197/800_review.patch

**Comment by codings-dan** [comment]
@szetszwo Thanks for reviewing the pull request, just have refactored the code, PTAL, thanks!

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by codings-dan** [comment]
@szetszwo Thanks a lot for helping review and merge the pull request!

**Pull request merged**